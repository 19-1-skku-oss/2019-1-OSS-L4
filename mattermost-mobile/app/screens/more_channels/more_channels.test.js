// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import MoreChannels from './more_channels.js';

jest.mock('react-intl');

describe('MoreChannels', () => {
    const navigator = {
        setOnNavigatorEvent: jest.fn(),
        setButtons: jest.fn(),
        dismissModal: jest.fn(),
        push: jest.fn(),
    };

    const actions = {
        handleSelectChannel: jest.fn(),
        joinChannel: jest.fn(),
        getChannels: jest.fn().mockResolvedValue({data: [{id: 'id2', name: 'name2', display_name: 'display_name2'}]}),
        searchChannels: jest.fn(),
        setChannelDisplayName: jest.fn(),
    };

    const baseProps = {
        actions,
        canCreateChannels: true,
        channels: [{id: 'id', name: 'name', display_name: 'display_name'}],
        closeButton: {},
        currentUserId: 'current_user_id',
        currentTeamId: 'current_team_id',
        navigator,
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <MoreChannels {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call props.navigator.dismissModal on close', () => {
        const wrapper = shallow(
            <MoreChannels {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.instance().close();
        expect(baseProps.navigator.dismissModal).toHaveBeenCalledTimes(1);
        expect(baseProps.navigator.dismissModal).toHaveBeenCalledWith({animationType: 'slide-down'});
    });

    test('should call props.navigator.setButtons on headerButtons', () => {
        const props = {...baseProps, navigator: {...navigator, setButtons: jest.fn()}};
        const wrapper = shallow(
            <MoreChannels {...props}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(props.navigator.setButtons).toHaveBeenCalledTimes(1);
        wrapper.instance().headerButtons(true);
        expect(props.navigator.setButtons).toHaveBeenCalledTimes(2);
    });

    test('should match return value of filterChannels', () => {
        const wrapper = shallow(
            <MoreChannels {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        const channels = [{id: 'id', name: 'name', display_name: 'display_name'}];
        expect(wrapper.instance().filterChannels(channels, 'name')).toEqual([channels[0]]);
        expect(wrapper.instance().filterChannels(channels, 'display_name')).toEqual([channels[0]]);
        expect(wrapper.instance().filterChannels(channels, 'none')).toEqual([]);
        expect(wrapper.instance().filterChannels(channels, 'display_none')).toEqual([]);
    });

    test('should match state on cancelSearch', () => {
        const wrapper = shallow(
            <MoreChannels {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.setState({term: 'term'});
        wrapper.instance().cancelSearch(true);
        expect(wrapper.state('term')).toEqual('');
        expect(wrapper.state('channels')).toEqual(baseProps.channels);
    });
});
