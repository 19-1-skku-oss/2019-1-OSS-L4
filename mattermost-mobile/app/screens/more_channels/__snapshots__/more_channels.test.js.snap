// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreChannels should match snapshot 1`] = `
<ForwardRef(forwardConnectRef)>
  <Connect(StatusBar) />
  <React.Fragment>
    <View
      style={
        Object {
          "marginVertical": 5,
        }
      }
    >
      <SearchBarIos
        autoCapitalize="none"
        backgroundColor="transparent"
        blurOnSubmit={true}
        inputHeight={33}
        inputStyle={
          Object {
            "backgroundColor": "rgba(61,60,64,0.2)",
            "color": "#3d3c40",
            "fontSize": 15,
          }
        }
        leftComponent={null}
        onBlur={[Function]}
        onCancelButtonPress={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSearchButtonPress={[Function]}
        onSelectionChange={[Function]}
        placeholderTextColor="rgba(61,60,64,0.5)"
        searchIconCollapsedMargin={10}
        searchIconExpandedMargin={10}
        tintColorDelete="rgba(61,60,64,0.5)"
        tintColorSearch="rgba(61,60,64,0.5)"
        titleCancelColor="#3d3c40"
        value=""
      />
    </View>
    <CustomList
      data={
        Array [
          Object {
            "display_name": "display_name",
            "id": "id",
            "name": "name",
          },
        ]
      }
      extraData={true}
      listType="flat"
      loading={true}
      loadingComponent={
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "height": 70,
              "justifyContent": "center",
            }
          }
        >
          <FormattedText
            defaultMessage="Loading Channels..."
            id="mobile.loading_channels"
            style={
              Object {
                "color": "rgba(61,60,64,0.6)",
              }
            }
          />
        </View>
      }
      noResults={null}
      onLoadMore={[Function]}
      onRowPress={[Function]}
      renderItem={[Function]}
      shouldRenderSeparator={true}
      showNoResults={true}
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
    />
  </React.Fragment>
</ForwardRef(forwardConnectRef)>
`;
